src/gui/src/viewer/zoom-window.h

```diff
@@ -8,13 +8,63 @@
 #include "downloader/image-save-result.h"
 #include "models/favorite.h"
 #include "models/image.h"
+#include "custom-buttons.h"
 
 
 namespace Ui
 {
 	class ZoomWindow;
-}
 
+	enum SaveButtonState : unsigned short
+	{
+		Save,
+		Saving,
+		Saved,
+		Copied,
+		Moved,
+		Linked,
+		ExistsMd5,
+		ExistsDisk,
+		Delete
+	};
```

Comment:
This SaveButtonState is a duplicate of the one in ZoomWindow. I did not remove the other or change the references to it because I am not sure whether you actually want to use the Ui namespace like this.